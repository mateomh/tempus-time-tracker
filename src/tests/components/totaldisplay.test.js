import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import TotalDisplay from '../../components/totaldisplay';
import rootReducer from '../../reducers/root';

describe('TotalDisplay Component', () => {
  let mockStore;

  beforeEach(() => {
    const user = {
      name: 'Testy',
    };

    const mockState = {
      sessionUser: user,
      filter: {},
      categories: [],
      tasks: [],
      total: 123456987,
      error: '',
    };

    mockStore = createStore(rootReducer, mockState);
  });

  it(
    '1. Renders correctly',
    () => {
      render(
        <Provider store={mockStore}>
          <TotalDisplay />
        </Provider>,
      );
      let item = screen.getByText(/Testy/);
      expect(item).toBeDefined();
      item = screen.getByText(/[123456987]/);
      expect(item).toBeDefined();
    },
  );

  it(
    '2. Checks it renders correctly against the snapshot',
    () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <TotalDisplay />
          </Provider>,
        ).toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
