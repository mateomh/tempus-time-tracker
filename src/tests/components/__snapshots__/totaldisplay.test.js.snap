// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalDisplay Component 2. Checks it renders correctly against the snapshot 1`] = `
<div
  className="Display"
>
  <p
    className="Text"
  >
    Testy
    ’s  total time for 
    All
     categories
  </p>
  <p
    className="Number"
  >
    123456987
     hours
  </p>
</div>
`;
